### PR TITLE
Add promise chaining for prompts

### PR DESCRIPTION
**Pull request by dhruvdutt**
Clean the nested promise waterfall approach and add promise chaining.

### CODE REVIEW

File: generator.js [hunk 1]
```diff
@@ -30,21 +30,21 @@ module.exports = class WebpackGenerator extends Generator {
...
+			})
+			.then(answer => {
+				if (answer['favicon']) {
+					// TODO: Add the default value here
+					return this.prompt([Input('favPath', 'Enter your fav icon path :')]);
+				}
+			})
+			.then(answer => {
+				if (answer) {
+					favPath = answer['favicon'];
```
Comment (sendilkumarn) [inline generator.js]: this should be `favPath`

Comment (dhruvdutt) [reply generator.js]: Fixed.

Comment (dhruvdutt) [reply generator.js]: I'll add the tests by tomorrow. Can we merge this?

**Comment by dhruvdutt** [comment]
@sendilkumarn We already have all the tests in place for things covered in this PR. I'll work on #3 in separate PR once this gets merged.

**Comment by sendilkumarn** [comment]
I would like to have few tests on prompts basically but that can be a separate issue 👍

**Pull request merged**